File: 2155-find-missing-observations/find-missing-observations.py
class Solution:
    def missingRolls(self, rolls: List[int], mean: int, n: int) -> List[int]:
        msum = sum(rolls)
        m = len(rolls)
        targetSum = mean * (m+n)
        leftSum = targetSum - msum
        
        res = []
        for i in range(n):
            v = min(6,max(1,(leftSum // (n-i)) + (leftSum % (n-i))))
            # print(leftSum,v)
            leftSum -= v
            res.append(v)
        if leftSum!=0:
            return []
        return res

File: 2155-find-missing-observations/find-missing-observations.java
class Solution {
    public int[] missingRolls(int[] rolls, int mean, int n) {
        int msum = 0;
        for (int i:rolls){
            msum+= i;
        }

        int targetSum = mean * (rolls.length+n);
        int leftSum = targetSum - msum;

        int[] res = new int[n];
        if ((leftSum / 6) > n){
            return new int[]{};
        }
        for (int i=0;i<n;i++){
            int q = leftSum / (n-i);
            int r = leftSum % (n-i);
            int v = Math.min(6, Math.max(1, q+r));
            leftSum -= v;
            res[i]=v;
        }
        if (leftSum != 0)
            return new int[]{};
        return res;
    }
}